context.go

```diff
@@ -135,18 +135,28 @@ func (ps *Set) GetAll() []*Plugin {
 // GetByType should be used. If only one is expected, then to switch plugins,
 // disable or remove the unused plugins of the same type.
 func (i *InitContext) GetSingle(t Type) (interface{}, error) {
-	pt, ok := i.plugins.byTypeAndID[t]
-	if !ok || len(pt) == 0 {
-		return nil, fmt.Errorf("no plugins registered for %s: %w", t, ErrPluginNotFound)
-	}
-	if len(pt) > 1 {
-		return nil, fmt.Errorf("multiple plugins registered for %s: %w", t, ErrPluginMultipleInstances)
+	var (
+		found    bool
+		instance interface{}
+	)
+	for _, v := range i.plugins.byTypeAndID[t] {
+		i, err := v.Instance()
+		if err != nil {
+			if IsSkipPlugin(err) {
+				continue
+			}
+			return instance, err
```

Comment:
why `return instance, err` instead of `return i, err`? is there some specific reason for doing this?

Reply:
Because I changed the name of the variable and forgot to change this one. I'll update it, it should be nil and not used but to be consistent.

Reply:
agree. thanks!